src/gpuarray_interface.jl

```diff
@@ -0,0 +1,110 @@
+import GPUArrays
+
+Base.similar(::Type{<: CuArray}, ::Type{T}, size::Base.Dims{N}) where {T, N} = CuArray{T, N}(size)
```

Comment:
This doesn't seem to have an analogue for other array types?

Reply:
When you want to generically offer constructors like CuArray(rand(10)) you only have a type and no instance.
This solves that problem ;) If it was up to me, I'd love to have this in Base being the only method to overload!